### PR TITLE
Add trace points for JITServer

### PR DESCRIPTION
**Pull request by a7ehuo**
Add the second set of trace points for JITServer.

Implements #9482

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by a7ehuo** [comment]
@mpirvu Ready for view. Thanks! There are two `TODO` comments around the tracepoints which will be addressed/removed in another PR that fixes the race condition around CHTable at the server.

File: runtime/compiler/env/j9jit.tdf [hunk 1]
```diff
@@ -90,26 +90,41 @@ TraceEvent=Trc_JITServerMethodSuccessfullyLoaded Overhead=1 Level=2 Test Templat
...
+TraceEvent=Trc_JITServerDoCHTableUpdate      Overhead=1 Level=1 Test Template="JITServer: compThreadID=%d clientSessionData=%p clientUID=%llu do CHTable update modified %llu and removed %llu"
+
+TraceEvent=Trc_JITServerUnloadClasses        Overhead=1 Level=1 Test Template="JITServer: compThreadID=%d clientSessionData=%p clientUID=%llu will process a list of %llu unloaded classes"
+TraceEvent=Trc_JITServerFreeCHTable          Overhead=1 Level=1 Test Template="JITServer: compThreadID=%d clientSessionData=%p clientUID=%llu will free CHTable"
 
 TraceEvent=Trc_JITServerStreamFailure                Overhead=1 Level=1 Test Template="JITServer: compThreadID=%d (%s) stream failure (compiling %s @ %s): %s"
 TraceEvent=Trc_JITServerStreamVersionIncompatible    Overhead=1 Level=1 Test Template="JITServer: compThreadID=%d (%s) stream version incompatible (compiling %s @ %s): %s"
 TraceEvent=Trc_JITServerStreamMessageTypeMismatch    Overhead=1 Level=1 Test Template="JITServer: compThreadID=%d (%s) stream message type mismatch (compiling %s @ %s): %s"
```
Comment (fjeremic) [inline runtime/compiler/env/j9jit.tdf:127]: Not sure if this is true any longer, or that we care about this for JITServer but I believe the trace point IDs of existing tracepoints will change if you add new tracepoints in the middle of one of these `.tdf` files. Probably doesn't matter here as I doubt anything is consuming these.

Comment (a7ehuo) [reply runtime/compiler/env/j9jit.tdf:127]: I think the trace point IDs of existing tracepoints will change in this case. I didn't find the existing code that consumes these IDs. So far all the new JITServer tracepoints have been added at the end of ` j9jit.tdf` which wouldn't affect the existing Non-JITServer tracepoint IDs. We're still rearranging the JITServer related tracepoints. Thanks for pointing this out. It's very good to keep it in mind when making the changes later on.

File: runtime/compiler/control/JITServerCompilationThread.cpp [hunk 2]
```diff
@@ -455,6 +476,12 @@ TR::CompilationInfoPerThreadRemote::processEntry(TR_MethodToBeCompiled &entry, J
       // Update the expecting sequence number. This will allow subsequent
       // threads to pass through once we've released the sequencing monitor.
       TR_ASSERT(seqNo == clientSession->getExpectedSeqNo(), "Unexpected seqNo=%u expected=%u\n", seqNo, clientSession->getExpectedSeqNo());
+
+      //TODO: Might need to set the expectedSeqNo as (seqNo+1) instead of (expectedSeqNo+1)
```
Comment (mpirvu) [inline runtime/compiler/control/JITServerCompilationThread.cpp:480]: This TODO may belong to another PR.

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit jdk11

**Comment by a7ehuo** [comment]
@mpirvu Tests have passed.
